src/client/dispatch.rs

```diff
@@ -70,357 +92,310 @@ pub(crate) fn dispatch_model(
 ///
 /// Can return `None` if an event is unknown.
 #[cfg_attr(not(feature = "cache"), allow(unused_mut))]
-fn update_cache_with_event(ctx: Context, event: Event) -> Option<(FullEvent, Option<FullEvent>)> {
+fn update_cache_with_event(
+    #[cfg(feature = "cache")] cache: &Cache,
+    event: Event,
+) -> Option<(FullEvent, Option<FullEvent>)> {
```

Comment:
It's overall a bad situation, but I don't think passing Context is any better, because it's still an undefined field access in that case.

Reply:
Couldn't you just inline the macro invocations and feature-flag them? Btw, I noticed there's calls to `if_cache!(event.update(cache))` which are actually equivalent to just `update_cache!(cache, event)`.

Reply:
`if_cache!(event.update(cache))` returns `None` if cache is disabled, whereas `update_cache!(cache, event)` returns `()`. This difference is needed because otherwise the `None` would need to be qualified for it's `T`.  I could of course inline the macro invocations, but that would get very repetitive very quickly and make it a lot harder to read.